src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1371,6 +1371,10 @@
     <Field Name="WhenExpression" Type="BoundExpression?"/>
     <Field Name="WhenTrue" Type="BoundDecisionDagNode" Null="disallow"/>
     <Field Name="WhenFalse" Type="BoundDecisionDagNode?"/>
+    <!--
+    Required is true when bindings are used in the opposite branch and cannot be optimized away.
+    -->
+    <Field Name="Required" Type="bool"/>
```

Comment:
We're going to need this for `or` pattern bindings when we split state for those as well.